packages/augur-core/source/contracts/reporting/RepPriceOracle.sol

```diff
@@ -45,73 +67,82 @@ contract RepPriceOracle is IRepPriceOracle, Initializable {
+        // The first time we have actual data from the exchange we need to simply record the acumulator value and return the default since we have no accurate _relative_ delta to base the real price off of
+        if (_exchangeData.repPriceAccumulated == 0) {
+            _exchangeData.blockNumber = _blockNumber;
+            _exchangeData.blockTimestamp = _blockTimestamp;
+            _exchangeData.repPriceAccumulated = _repPriceCumulative;
             return _exchangeData;
         }
-        uint256 _deltaPrice = _cashAccumulationDelta * 1 ether / _repAccumulationDelta;
 
-        uint256 _weight = getWeight(_blockNumber - _exchangeData.blockNumber);
-        uint256 _price = (((1 ether - _weight) * _exchangeData.price) + (_weight * _deltaPrice)) / 1 ether;
+        uint256 _blocksElapsed = _blockNumber - _exchangeData.blockNumber;
 
-        _exchangeData.price = _price;
-        _exchangeData.blockNumber = _blockNumber;
-        _exchangeData.cashAmountAccumulated = _cashAmountAccumulated;
-        _exchangeData.repAmountAccumulated = _repAmountAccumulated;
+        uint256 _price = (_repPriceCumulative - _exchangeData.repPriceAccumulated) * 10**18 / _blocksElapsed / Q112;
+        require(_price > 0, "Price should not be 0");
 
-        return _exchangeData;
-    }
+        uint256 _secondsElapsed = _blockTimestamp - _exchangeData.blockTimestamp;
+        uint256 _priceAverage = _price;
 
-    function getWeight(uint256 _blockDelta) public view returns (uint256) {
-        uint256 _weight = _blockDelta * 1 ether / tau;
-        if (_weight > 1 ether) {
-            return 1 ether;
+        if (_secondsElapsed < period) {
+            _priceAverage = (_exchangeData.price * (period - _secondsElapsed) + _price * _secondsElapsed) / period;
         }
-        return _weight;
+
+        _exchangeData.blockNumber = _blockNumber;
+        _exchangeData.blockTimestamp = _blockTimestamp;
+        _exchangeData.repPriceAccumulated = _repPriceCumulative;
+        _exchangeData.price = _priceAverage;
+        return _exchangeData;
     }
 
     function initializeUniverse(IV2ReputationToken _reputationToken) private {
-        exchangeData[address(_reputationToken)].exchange = getOrCreateUniswapExchange(_reputationToken);
+        uint256 _blockNumber = block.number;
+        IUniswapV2 _exchange = getOrCreateUniswapExchange(_reputationToken);
+        exchangeData[address(_reputationToken)].exchange = _exchange;
+        if (_blockNumber != _exchange.blockNumberLast()) {
+            _exchange.sync();
+        }
         uint256 _initialPrice = getInitialPrice(_reputationToken);
         exchangeData[address(_reputationToken)].price = _initialPrice;
-        exchangeData[address(_reputationToken)].blockNumber = block.number;
+        exchangeData[address(_reputationToken)].blockNumber = _blockNumber;
+        exchangeData[address(_reputationToken)].blockTimestamp = block.timestamp; // solium-disable-line security/no-block-members
```

Comment:
I'm curious why you don't just turn off that solium check?  I don't think it really adds value in this project since we use `block.timestamp` in a number of places, as well as `block.number`.

Reply:
We really only use block.timestamp in 3 places, but yea should probably just turn off that check.